docs/understanding-the-flutterflow-ui/projects/how-to-run-test-projects.md

```diff
@@ -0,0 +1,94 @@
+</div>
+<div class="alert alert--info" role="alert">
+<h4>Troubleshooting – Slow Loading</h4>
+If you see a progress bar at the top of the device running for over 15 seconds, try refreshing the page.
+</div>
+<div class="alert alert--warning" role="alert">
+#### Test Mode Limitations
+- 30 minute testing time. After the 30 minutes has expired, you can start another test mode.
+- If you see a grey "broken" screen with a sad face, it may be a DNS server issue with your network provider. We recommend using CloudFlare's 1.1.1.1 DNS server. Click here to see instructions.
+- Lottie animation may not load if you provide a variable path.
+- Cookies need to be enabled for Test Mode to function properly. They are only used for functional purposes.
+- If you see a progress bar where the phone outline should be that lasts longer than a 15 seconds, try refreshing the page.
+- The device screen can not be wider than the page's width.
+- Copy to Clipboard Action is not supported in Test Mode. Use Run Mode to avoid this issue.
+- Widgets having Shimmer or Tint animation might not appear properly.
+- Assets used within Custom Code might not appear properly.
+- Audio Recording actions do not work in Test Mode, use web publishing in Settings to test recording audio.
+</div>
+
+## Run Mode
+
+You can test a fully functional version of your app using the _Run_ mode, including live data. It will build the app, which typically requires around 2-4 minutes - but can be longer for larger projects. You can then interact with your app through your web browser.
+
+To run the app in Run Mode:
+
+1. Click on the **dropdown** next to the Test Mode button \***\*or press **Cmd/Ctrl + E\*\* (a keyboard shortcut). This will run your app in a new browser window.
+<div class="alert alert--info" role="alert">
+#### When to Use Run Mode
+The main benefit of Run Mode is the ability to share a running app. The url of the Run Mode is public and shareable. All Run Modes will persist and can be accessed in the drop down menu next to the lightning bolt icon in upper right of the FlutterFlow builder.
+</div>
+<div class="alert alert--warning" role="alert">
+#### Run Mode Limitations
+Run Mode does not support Hot Reloading so any changes you make to your app will not be reflected in the Run Mode. In order to see the changes you would have to create another Run Mode.
+</div>
+
+## Local Run
+You can test your app on an emulator or physical mobile device with Local Run feature, which is available in the FlutterFlow Desktop App. Local Run automatically tracks changes in your FlutterFlow project, downloads the code locally, and gives you the option to use Flutter's Hot Reload or Hot Restart to see your changes instantly on a device.
```

Comment:
"Local Run automatically tracks changes in your FlutterFlow project" - I didn't quite get this. What do we mean by this?

Reply:
Yeah, maybe unnecesssary. I think the point was that changes are automatically saved.